plugins/woocommerce/includes/wc-ipp-functions.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * WooCommerce In-Person Payments functions
+ *
+ * @package WooCommerce\Functions
+ * @version 7.0
+ */
+
+/**
+ * Returns if order is eligible to accept In-Person Payments
+ *
+ * @param WC_Order $order order that the conditions are checked for.
+ *
+ * @return bool true if order is eligible, false otherwise
+ */
+function is_order_in_person_payment_eligible( WC_Order $order ): bool {
+	$has_status            = in_array( $order->get_status(), array( 'pending', 'on-hold', 'processing' ), true );
+	$has_payment_method    = in_array( $order->get_payment_method(), array( 'cod', 'woocommerce_payments', 'none' ), true );
+	$order_is_not_paid     = null === $order->get_date_paid();
+	$order_is_not_refunded = empty( $order->get_refunds() );
+
+	$order_has_no_subscription_products = true;
+	foreach ( $order->get_items() as $item ) {
+		$product = $item->get_product();
+
+		if ( $product->is_type( 'subscription' ) ) {
```

Comment:
Something to consider:

```suggestion
		$product = $item->get_product();

		if ( is_object( $product ) && $product->is_type( 'subscription' ) ) {
```

Though it is presumably unusual, `get_items()` is documented as possibly returning `bool`.

Reply:
Just to better understand: if `get_product()` would return `bool`, the `is_type` executed on it would cause crash, right?

Reply:
Exactly ... we'd get a fatal error. 

I don't know for sure that there is a chain of events that would lead to us actually receiving `(bool) false` at this point—but the *potential* is there, so being defensive is something to consider.